mypy_django_plugin/transformers/managers.py

```diff
@@ -244,7 +245,10 @@ def _replace_type_var(ret_type: MypyType, to_replace: str, replace_by: MypyType)
     """
     if isinstance(ret_type, TypeVarType) and ret_type.fullname == to_replace:
         return replace_by
-    elif isinstance((instance := get_proper_type(ret_type)), Instance):
+
+    ret_type = copy_type(get_proper_type(ret_type))
+
+    if isinstance((instance := ret_type), Instance):
```

Comment:
No need to rename any longer here IMO

```suggestion
    if isinstance(ret_type, Instance):
```

Reply:
👍